connexion/middleware/routing.py

```diff
@@ -129,3 +136,46 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
         # Needs to be set so starlette router throws exceptions instead of returning error responses
         scope["app"] = self
         await self.router(scope, receive, send)
+
+
+PATH_PARAMETER = re.compile(r"\{([^}]*)\}")
+PATH_PARAMETER_CONVERTERS = {"integer": "int", "number": "float", "path": "path"}
+
+
+def convert_path_parameter(match, types):
+    name = match.group(1)
+    swagger_type = types.get(name)
+    converter = PATH_PARAMETER_CONVERTERS.get(swagger_type)
+    return f'{{{name.replace("-", "_")}{":" if converter else ""}{converter or ""}}}'
+
+
+def starlettify_path(swagger_path, types=None):
```

Comment:
This fixes routing for conflicting paths with path parameters with different types.